src/ui/common/src/components/tables/virtualizedTable.tsx

```diff
@@ -0,0 +1,252 @@
+import { Box, Typography } from '@mui/material';
+import { styled, Theme } from '@mui/material/styles';
+import TableCell from '@mui/material/TableCell';
+import clsx from 'clsx';
+import * as React from 'react';
+import {
+  AutoSizer,
+  Column,
+  Table,
+  TableCellRenderer,
+  TableHeaderProps,
+} from 'react-virtualized';
+
+const classes = {
+  flexContainer: 'ReactVirtualizedDemo-flexContainer',
+  tableRow: 'ReactVirtualizedDemo-tableRow',
+  tableRowHover: 'ReactVirtualizedDemo-tableRowHover',
+  tableCell: 'ReactVirtualizedDemo-tableCell',
+  noClick: 'ReactVirtualizedDemo-noClick',
+};
+
+const columnWidthMultiplier = 13;
+
+const styles = ({ theme }: { theme: Theme }) =>
+  ({
+    // temporary right-to-left patch, waiting for
+    // https://github.com/bvaughn/react-virtualized/issues/454
+    '& .ReactVirtualized__Table__headerRow': {
+      ...(theme.direction === 'rtl' && {
+        paddingLeft: '0 !important',
+      }),
+      ...(theme.direction !== 'rtl' && {
+        paddingRight: undefined,
+      }),
+    },
+    '& .ReactVirtualized__Table__headerColumn': {
+      ...{
+        marginRight: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__rowColumn': {
+      ...{
+        marginRight: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__headerColumn:first-of-type': {
+      ...{
+        marginLeft: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__rowColumn:first-of-type': {
+      ...{
+        marginLeft: '0px',
+      },
+    },
+    [`& .${classes.flexContainer}`]: {
+      display: 'flex',
+      alignItems: 'center',
+      boxSizing: 'border-box',
+    },
+    [`& .${classes.tableRow}`]: {
+      cursor: 'pointer',
+    },
+    [`& .${classes.tableRowHover}`]: {
+      '&:hover': {
+        backgroundColor: theme.palette.grey[200],
+      },
+    },
+    [`& .${classes.tableCell}`]: {
+      flex: 1,
+    },
+    [`& .${classes.noClick}`]: {
+      cursor: 'initial',
+    },
+  } as const);
+
+interface ColumnData {
+  dataKey: string;
+  label: string;
+  type: string;
+  numeric?: boolean;
+  columnWidth?: number;
+}
+
+interface Row {
+  index: number;
+}
+
+interface MuiVirtualizedTableProps {
+  columns: readonly ColumnData[];
+  headerHeight?: number;
+  minColumnWidth?: number;
+  onRowClick?: () => void;
+  rowCount: number;
+  rowGetter: (row: Row) => any;
+  rowHeight?: number;
+}
+
+class MuiVirtualizedTable extends React.PureComponent<MuiVirtualizedTableProps> {
```

Comment:
Style nit: The fact that the table is virtualized seems like an implementation detail, so I think we should still call this a `DataTable`. 

Also, I think the general style we've been trying to follow is that both the class names and file names should be PascalCased. So here, I think we should call both `DataTable`.

Reply:
At the end of this file we have: `const DataTable = styled(<ClassName>)(styles);`. To prevent name collision I'm naming the class `PreStyledDataTable`. Let me know if you have better ideas. @agiron123 @vsreekanti

Reply:
That's fine with me!

Reply:
@vsreekanti @cw75 I'd rather leave this one named VirtualizedTable and have DataTable be the parent class that all other tables inherit from. (more of an OOP analogy here, but you get the point)

WIth that being said, I have a table view that I am working on right now that will have pagination. 

After that goes in, we can combine all three into one table component and control which table component is used based on the props that we pass in.

let's leave as is for now and then we can revisit and build out a nicer DataTable implementation afterwards.